3Dmol/parsers.js

```diff
@@ -2202,5 +2202,52 @@ $3Dmol.Parsers = (function() {
 	}
         return atoms;
     }
+
+    /**
+     * Parse a lammps trajectory file from str and create atoms
+     */
+    parsers.lammpstrj = parsers.LAMMPSTRJ = function(str, options){
+ 	var atoms = [];
+        var dic = {'id':'serial','xs':'x','ys':'y','zs':'z'};
+        var lines = str.split(/\r?\n|\r/);
+        while (lines.length > 0){
+            if (lines.length < 9)
+                break;
+            var offset = 0;
+            var atomCount = 0;
+            for (var i=0; i<lines.length; i++){
+                if (lines[i].match(/ITEM: ATOMS/)){
+                    offset = i+1;
+                    break;
+                }
+                if (lines[i].match(/ITEM: NUMBER OF ATOMS/))
+                    atomCount = parseInt(lines[i+1]);
+            }
+            var types = lines[offset-1].replace('ITEM: ATOMS ','').split(' ');
+            atoms.push([]);
+            for (var i=offset; i<offset+atomCount; i++){
+                var atom = {};
+                var tokens = lines[i].split(' ');
+                for (var j=0; j<tokens.length; j++){
+                    var prop = dic[types[j]];
+                    if (prop != undefined){
+                        if (prop == 'serial')
+                            atom[prop] = parseInt(tokens[j]);
+                        else
+                            atom[prop] = parseFloat(tokens[j]);
+                     }
+                     atom.bonds = [];
+                     atom.bondOrder = [];
+                     atom.properties = {};
+                     atoms[atoms.length-1][i-offset] = atom;   
+                }
+            }
+            lines.splice(0,offset+atomCount-1);
```

Comment:
splicing isn't particularly efficient as it has to construct a new array.  You should avoid modifying the array.  You can have a nested loop structure where the inner loop updates the iteration index of the outer loop (pseudocode):
```
for(var i = 0, n = lines.length; i < n; i++ ) {
 if (lines[i].match(/ITEM: NUMBER OF ATOMS/))
                   atomCount = parseInt(lines[i+1]);
else if (lines[i].match(/ITEM: ATOMS/)) {
   for(var stop = i + atomCount; i < stop; i++) {
   ...
  }
}
```